api/session.go

```diff
@@ -179,6 +180,10 @@ func (s *Session) RenewPeriodic(initialTTL string, id string, q *WriteOptions, d
 			// Attempt a session destroy
 			s.Destroy(id, q)
 			return nil
+		case <-ctx.Done():
+			// Attempt a session destroy
+			s.Destroy(id, q)
+			return ctx.Err()
 		}
 	}
 }
```

Comment:
Could you make a simple unit test similar to  https://github.com/hashicorp/consul/blob/7bbfef7000be7ba163534c281369c8c6f8bfaea4/api/session_test.go#L185 but with a context instead? After that this looks close to being ready to merge.